src/main/kotlin/parts/wisdom/arcraider/visualization/ARCWindow.kt

```diff
@@ -1,56 +1,143 @@
 package parts.wisdom.arcraider.visualization
 
+import parts.wisdom.arcraider.SerializedGrid
 import parts.wisdom.arcraider.Task
 import parts.wisdom.arcraider.loadTaskFromFile
 import java.awt.Dimension
-import java.awt.FlowLayout
-import javax.swing.*
+import javax.swing.Box
+import javax.swing.BoxLayout
+import javax.swing.JButton
+import javax.swing.JFileChooser
+import javax.swing.JFrame
+import javax.swing.JLabel
+import javax.swing.JPanel
+import javax.swing.JToggleButton
 
-class ARCWindow(width : Int, height : Int, title : String, task : Task) : JFrame() {
+private const val GAP_SIZE = 12
+private const val GRID_SIZE = 40
 
-    private lateinit var gridPane : GridPane
+typealias TaskGrids = Pair<List<GridPane>, List<GridPane>>
```

Comment:
This doesn't fit the logical structure of an ARC task, which has pairs of (input grid, output grid). The logical structure becomes important visually when grid sizes vary. Here's an [example](https://snipboard.io/D7c340.jpg).